### PR TITLE
[MPIR-460] Dependency Information for maven-plugin

### PR DESCRIPTION
**Pull request by slawekjaranowski**
https://issues.apache.org/jira/browse/MPIR-460

### CODE REVIEW

File: src/it/MPIR-460-jar/verify.groovy [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+def report = new File( basedir, 'target/site/dependency-info.html' ).text
+
+assert !report.contains('<pre>&lt;plugin&gt;')
+assert !report.contains('&lt;/plugin&gt;</pre>')
+
+assert report.contains('<pre>&lt;dependency&gt;')
+assert report.contains('&lt;/dependency&gt;</pre>')
+
+assert report.contains('Apache Ivy')
+assert report.contains('Groovy Grap')
```
Comment (michael-o) [inline src/it/MPIR-460-jar/verify.groovy:28]: Grape, not Grap

File: src/it/MPIR-460-jar/verify.groovy [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+assert report.contains('Gradle/Grail')
```
Comment (michael-o) [inline src/it/MPIR-460-jar/verify.groovy:29]: Grails

File: src/it/MPIR-460-jar/pom.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.apache.maven.plugins.project-info-reports</groupId>
+  <artifactId>MPIR-460-jar</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>jar</packaging>
+
+  <name>MPIR-460 IT test: dependency-info for jar packaging</name>
+  <description>
+    Coordinate list are generated for many tools
```
Comment (michael-o) [inline src/it/MPIR-460-jar/pom.xml:33]: Coordinates

File: src/it/MPIR-460-maven-plugin/pom.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.apache.maven.plugins.project-info-reports</groupId>
+  <artifactId>MPIR-460-maven-plugin</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>maven-plugin</packaging>
+
+  <name>MPIR-460 IT test: dependency-info for maven-plugin</name>
+  <description>
+    Only coordinate for Maven plugin is generated
```
Comment (michael-o) [inline src/it/MPIR-460-maven-plugin/pom.xml:33]: Coordinates

File: src/it/MPIR-460-maven-plugin/verify.groovy [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+def report = new File( basedir, 'target/site/dependency-info.html' ).text
+
+assert report.contains('<pre>&lt;plugin&gt;')
+assert report.contains('&lt;/plugin&gt;</pre>')
+
+assert !report.contains('<pre>&lt;dependency&gt;')
+assert !report.contains('&lt;/dependency&gt;</pre>')
+
+assert !report.contains('Apache Ivy')
+assert !report.contains('Groovy Grap')
```
Comment (michael-o) [inline src/it/MPIR-460-maven-plugin/verify.groovy:28]: Same here

File: src/it/MPIR-460-maven-plugin/verify.groovy [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+assert !report.contains('Gradle/Grail')
```
Comment (michael-o) [inline src/it/MPIR-460-maven-plugin/verify.groovy:29]: Same here

File: src/main/java/org/apache/maven/report/projectinfo/DependencyInformationReport.java [hunk 7]
```diff
@@ -130,6 +131,32 @@ protected String getI18Nsection() {
...
+            } else {
+                renderMavenDependencyCoordinate();
+                renderIvyDependencyCoordinate();
+                renderGrapeDependencyCoordinate();
+                renderGradleDependencyCoordinate();
+                renderScalaDependencyCoordinate();
+                renderLeiningenDependencyCoordinate();
+            }
+
+            endSection();
```
Comment (michael-o) [inline src/main/java/org/apache/maven/report/projectinfo/DependencyInformationReport.java:145]: `...Coordinates()`

**Review by michael-o — CHANGES_REQUESTED** [review]

**Review by michael-o — APPROVED** [review]
Looks totally reasonable to me.

**Comment by jira-importer** [comment]
Resolve #350

**Pull request merged**